src/cljs-lib/src/calva/lsp.cljs

```diff
@@ -0,0 +1,220 @@
+  ;; The title of this command is dictated by clojure-lsp and is executed when the user clicks the references code lens above a symbol
+  (.. context -subscriptions
+      (push (.. vscode -commands
+                (registerCommand "code-lens-references" code-lens-references-callback))))
+  (run! (fn [command]
+          (register-lsp-command client command))
+        lsp-commands))
+
+(defn handle-toggle-references-code-lens
+  [^js event]
+  (when (.. event (affectsConfiguration "calva.referencesCodeLens.enabled"))
+    (let [visible-editors (filter 
+                           (fn [editor]
+                             (= (.. editor -document -uri -scheme) "file"))
+                           (.. vscode -window -visibleTextEditors))]
+      ;; Hacky solution for triggering codeLens refresh
+      ;; Could not find a better way, aside from possibly changes to clojure-lsp
+      ;; https://github.com/microsoft/vscode-languageserver-node/issues/705
+      (run! (fn [editor]
+              (js/console.log "adding edits for:" (.. editor -document -uri toString))
+              (let [edit1 ^js (new (.. vscode -WorkspaceEdit))
+                    _ (.. edit1 (insert (.. editor -document -uri)
+                                        (new (.. vscode -Position) 0 0)
+                                        "\n"))
+                    edit2 ^js (new (.. vscode -WorkspaceEdit))
+                    _ (.. edit2 (delete (.. editor -document -uri)
+                                        (new (.. vscode -Range) 0 0 1 0)))]
+                (go
+                  (<p! (.. vscode -workspace (applyEdit edit1)))
+                  (<p! (.. vscode -workspace (applyEdit edit2))))))
+            visible-editors))))
+
+(defn register-event-handlers
+  [^js context]
+  (.. context -subscriptions
+      (push (.. vscode -workspace
+                (onDidChangeConfiguration handle-toggle-references-code-lens)))))
+
+(defn activate [^js context]
+  (let [jar-path (. path join (. context -extensionPath) "clojure-lsp.jar")
+        client (create-client jar-path)]
+    (. client start)
+    (.. vscode -window
+        (setStatusBarMessage
+         "$(sync~spin) Initializing Clojure language features via clojure-lsp"
+         (. client onReady)))
+    (.. (. client onReady)
+        (then #(do
+                 (.. state -cursor (set client-key client))
+                 (register-commands context client)
+                 (register-event-handlers context))))))
+
+(defn deactivate []
+  (when-let [client (.. state deref (get client-key))]
+    (.. client stop)))
+
+(defn get-references
+  [^LanguageClient client uri ^Position position include-declaration]
+  (.. client
+      (sendRequest "textDocument/references"
+                   (clj->js {:textDocument {:uri uri}
+                             :position position
+                             :context {:includeDeclaration 
+                                       (if (nil? include-declaration)
```

Comment:
Refactor to multi-arity function, passing true for this param. See if this works when called from js.